### PR TITLE
[hot reload][mono] Implement support for adding static and instance fields to generic classes

### LINKED ISSUE #86111: [hot reload] Support the GenericAddFieldToExistingType capability in Mono

### PR DESCRIPTION
**Pull request by lambdageek**
Fixes https://github.com/dotnet/runtime/issues/86111

### CODE REVIEW

**Comment by lambdageek** [comment]
cc @mikelle-rogers coreclr doesn't like the new test

**Comment by lambdageek** [comment]
Also found an issue where mono does loads of added valuetype fields incorrectly.  That warrants its own non-generic test (the existing tests only try small primitive types, not structs like `DateTime`).

**Comment by lambdageek** [comment]
@fanyang-mono please take a look

**Review by fanyang-mono — APPROVED** [review]
LGTM! I wonder if you need to update `hot_reload_get_capabilities` for this new capability?

**Comment by lambdageek** [comment]
> LGTM! I wonder if you need to update `hot_reload_get_capabilities` for this new capability?

yes! thank you!

File: src/mono/mono/mini/interp/transform.c [hunk 1]
```diff
@@ -6571,7 +6571,11 @@ generate_code (TransformData *td, MonoMethod *method, MonoMethodHeader *header,
 						MonoClass *field_class = mono_class_from_mono_type_internal (ftype);
 						interp_emit_metadata_update_ldflda (td, field, error);
 						goto_if_nok (error, exit);
```
Comment (BrzVlad) [inline src/mono/mono/mini/interp/transform.c:6573]: Looks like the whole code following here should be replaced with `interp_emit_ldobj`

Comment (lambdageek) [reply src/mono/mono/mini/interp/transform.c:6573]: @BrzVlad thanks! I also added `BARRIER_IF_VOLATILE`

**Review by BrzVlad — APPROVED** [review]

**Pull request merged**